lib/absinthe/blueprint/input.ex

```diff
@@ -65,5 +67,25 @@ defmodule Absinthe.Blueprint.Input do
     nil
   end
 
+  @simple_inspect_types [
+    Input.Boolean,
+    Input.Float,
+    Input.Integer,
+    Input.String
+  ]
+
+  @spec inspect(t) :: String.t
+  def inspect(%str{value: value}) when str in @simple_inspect_types do
+    Kernel.inspect(value)
+  end
+  def inspect(%Input.Enum{value: value}) do
+    value
+  end
+  def inspect(%Input.List{values: values}) do
+    contents = values
+    |> Enum.map(&inspect/1)
+    |> Enum.join(", ")
+    "[#{contents}]"
+  end
```

Comment:
No; perhaps I should call it something else, but I don't want to implement
Inspect and screw up debugging elsewhere.
On Sat, Aug 27, 2016 at 3:23 PM Ben Wilson <notifications@github.com> wrote:

> In lib/absinthe/blueprint/input.ex
> <https://github.com/absinthe-graphql/absinthe/pull/130#discussion_r76523145>
> :
>
> > +    Input.String
> > +  ]
> > +
> > +  @spec inspect(t) :: String.t
> > +  def inspect(%str{value: value}) when str in @simple_inspect_types do
> > +    Kernel.inspect(value)
> > +  end
> > +  def inspect(%Input.Enum{value: value}) do
> > +    value
> > +  end
> > +  def inspect(%Input.List{values: values}) do
> > +    contents = values
> > +    |> Enum.map(&inspect/1)
> > +    |> Enum.join(", ")
> > +    "[#{contents}]"
> > +  end
>
> @bruce <https://github.com/bruce> do we want to just implement the
> inspect protocol?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/absinthe-graphql/absinthe/pull/130/files/894ae7c46d4cb85e4df65397ffbd31e64728ffd4#r76523145>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAASDjWFGp6Kos41W872Krh701jzWRrks5qkLj2gaJpZM4JSUdL>
> .
>